MapleServer2/PacketHandlers/Game/GuildHandler.cs

```diff
@@ -437,11 +477,29 @@ private void HandleEnterHouse(GameSession session, PacketReader packet)
         private void HandleGuildDonate(GameSession session, PacketReader packet)
         {
             int donateQuantity = packet.ReadInt();
+            int donationAmount = donateQuantity * 10000;
+            Guild guild = GameServer.GuildManager.GetGuildById(session.Player.GuildId);
+            if (guild == null)
+            {
+                return;
+            }
+
+            if (!session.Player.Wallet.Meso.Modify(-donationAmount))
+            {
+                short NoticeCode = 5121;
+                session.Send(GuildPacket.ErrorNotice(NoticeCode));
+                return;
+            }
+            GuildContribution contribution = GuildMetadataStorage.GetContribution("donation");
+
+            session.Player.GuildContribution += contribution.Value * donateQuantity;
+            int guildFunds = guild.Funds + donateQuantity * 10000;
```

Comment:
You can replace `donateQuantity * 10000` with `donationAmount` here since it's already calculated for you